<a href="https://colab.research.google.com/github/ypjuri1728/EDA-ML-Journey/blob/main/Titanic_EDA_Day01.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
df = pd.read_csv("tested.csv")

In [ ]:
df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,892,0,3,"Kelly, Mr. James",male,34.5,0,0,330911,7.8292,NaN,Q
1,893,1,3,"Wilkes, Mrs. James (Ellen Needs)",female,47.0,1,0,363272,7.0000,NaN,S
2,894,0,2,"Myles, Mr. Thomas Francis",male,62.0,0,0,240276,9.6875,NaN,Q
3,895,0,3,"Wirz, Mr. Albert",male,27.0,0,0,315154,8.6625,NaN,S
4,896,1,3,"Hirvonen, Mrs. Alexander (Helga E Lindqvist)",female,22.0,1,1,3101298,12.2875,NaN,S


In [ ]:
df.shape

(418, 12)

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 418 entries, 0 to 417
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  418 non-null    int64  
 1   Survived     418 non-null    int64  
 2   Pclass       418 non-null    int64  
 3   Name         418 non-null    object 
 4   Sex          418 non-null    object 
 5   Age          332 non-null    float64
 6   SibSp        418 non-null    int64  
 7   Parch        418 non-null    int64  
 8   Ticket       418 non-null    object 
 9   Fare         417 non-null    float64
 10  Cabin        91 non-null     object 
 11  Embarked     418 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 39.3+ KB


In [ ]:
df.describe()

,PassengerId,Survived,Pclass,Age,SibSp,Parch,Fare
count,418.000000,418.000000,418.000000,332.000000,418.000000,418.000000,417.000000
mean,1100.500000,0.363636,2.265550,30.272590,0.447368,0.392344,35.627188
std,120.810458,0.481622,0.841838,14.181209,0.896760,0.981429,55.907576
min,892.000000,0.000000,1.000000,0.170000,0.000000,0.000000,0.000000
25%,996.250000,0.000000,1.000000,21.000000,0.000000,0.000000,7.895800
50%,1100.500000,0.000000,3.000000,27.000000,0.000000,0.000000,14.454200
75%,1204.750000,1.000000,3.000000,39.000000,1.000000,0.000000,31.500000
max,1309.000000,1.000000,3.000000,76.000000,8.000000,9.000000,512.329200


In [ ]:
df.isnull().sum()

,0
PassengerId,0
Survived,0
Pclass,0
Name,0
Sex,0
Age,86
SibSp,0
Parch,0
Ticket,0
Fare,1


In [ ]:
df.nunique()

,0
PassengerId,418
Survived,2
Pclass,3
Name,418
Sex,2
Age,79
SibSp,7
Parch,8
Ticket,363
Fare,169


In [ ]:
df["Sex"].value_counts()

,count
Sex,
male,266
female,152


In [ ]:
df["Pclass"].value_counts()

,count
Pclass,
3,218
1,107
2,93


In [ ]:
df["Embarked"].value_counts()

,count
Embarked,
S,270
C,102
Q,46


In [ ]:
df["Survived"].value_counts()

,count
Survived,
0,266
1,152


whatever we use this valuecount for the understand what columns actually contains.

In [ ]:
df.duplicated().sum()

np.int64(0)

asking: Are there duplicates rows in my dataset?

In [ ]:
df.describe()

,PassengerId,Survived,Pclass,Age,SibSp,Parch,Fare
count,418.000000,418.000000,418.000000,332.000000,418.000000,418.000000,417.000000
mean,1100.500000,0.363636,2.265550,30.272590,0.447368,0.392344,35.627188
std,120.810458,0.481622,0.841838,14.181209,0.896760,0.981429,55.907576
min,892.000000,0.000000,1.000000,0.170000,0.000000,0.000000,0.000000
25%,996.250000,0.000000,1.000000,21.000000,0.000000,0.000000,7.895800
50%,1100.500000,0.000000,3.000000,27.000000,0.000000,0.000000,14.454200
75%,1204.750000,1.000000,3.000000,39.000000,1.000000,0.000000,31.500000
max,1309.000000,1.000000,3.000000,76.000000,8.000000,9.000000,512.329200


In [ ]:
df[df["Fare"] > 100][["PassengerId", "Pclass", "Fare"]]

,PassengerId,Pclass,Fare
24,916,1,262.3750
53,945,1,263.0000
59,951,1,262.3750
64,956,1,262.3750
69,961,1,263.0000
74,966,1,211.5000
75,967,1,211.5000
81,973,1,221.7792
114,1006,1,221.7792
141,1033,1,151.5500


In [ ]:
df.groupby("Pclass")["Fare"].agg(["count", "mean", "median", "min", "max"])

,count,mean,median,min,max
Pclass,,,,,
1,107,94.280297,60.0000,0.0000,512.3292
2,93,22.202104,15.7500,9.6875,73.5000
3,217,12.459678,7.8958,3.1708,69.5500


In [ ]:
df[(df["Pclass"] == 1) & (df["Fare"] == df["Fare"].max())]

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
343,1235,1,1,"Cardeza, Mrs. James Warburton Martinez (Charlo...",female,58.0,0,1,PC 17755,512.3292,B51 B53 B55,C


Pclass 1 -> higher fares

Pclass 2 -> medium fares

Pclass 3 -> lower fares

The highest fare belongs to a first-class passenger, so the extreme fare appears to be associated with passenger class rather than automatically being an invalid value.

In [ ]:
df.groupby("Pclass")["Survived"].mean()

,Survived
Pclass,
1,0.467290
2,0.322581
3,0.330275


In [ ]:
df[df["Pclass"] == 1][["PassengerId", "Pclass", "Fare"]].sort_values("Fare", ascending=False).head(10)

,PassengerId,Pclass,Fare
343,1235,1,512.3292
69,961,1,263.0000
53,945,1,263.0000
59,951,1,262.3750
64,956,1,262.3750
142,1034,1,262.3750
24,916,1,262.3750
375,1267,1,262.3750
184,1076,1,247.5208
202,1094,1,227.5250


In [ ]:
df[df["Fare"] > 200][
    ["PassengerId", "Name", "Sex", "Age", "Pclass", "Fare", "Ticket", "Cabin", "SibSp", "Parch"]
]

,PassengerId,Name,Sex,Age,Pclass,Fare,Ticket,Cabin,SibSp,Parch
24,916,"Ryerson, Mrs. Arthur Larned (Emily Maria Borie)",female,48.0,1,262.3750,PC 17608,B57 B59 B63 B66,1,3
53,945,"Fortune, Miss. Ethel Flora",female,28.0,1,263.0000,19950,C23 C25 C27,3,2
59,951,"Chaudanson, Miss. Victorine",female,36.0,1,262.3750,PC 17608,B61,0,0
64,956,"Ryerson, Master. John Borie",male,13.0,1,262.3750,PC 17608,B57 B59 B63 B66,2,2
69,961,"Fortune, Mrs. Mark (Mary McDougald)",female,60.0,1,263.0000,19950,C23 C25 C27,1,4
74,966,"Geiger, Miss. Amalie",female,35.0,1,211.5000,113503,C130,0,0
75,967,"Keeping, Mr. Edwin",male,32.5,1,211.5000,113503,C132,0,0
81,973,"Straus, Mr. Isidor",male,67.0,1,221.7792,PC 17483,C55 C57,1,0
114,1006,"Straus, Mrs. Isidor (Rosalie Ida Blun)",female,63.0,1,221.7792,PC 17483,C55 C57,1,0
142,1034,"Ryerson, Mr. Arthur Larned",male,61.0,1,262.3750,PC 17608,B57 B59 B63 B66,1,3


The Fare is likely related to the booking/ticket, not simply an individual passenger.

512.3292 is still unusually high, but we now have evidence that high fares can be legitimate ticket-level values.

In [ ]:
df[df["Ticket"] == "PC 17755"]

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
343,1235,1,1,"Cardeza, Mrs. James Warburton Martinez (Charlo...",female,58.0,0,1,PC 17755,512.3292,B51 B53 B55,C


**Investigation: Checked high-fare passengers, compared fares by Pclass, and examined ticket patterns.**

*Insight: High fares are strongly associated with first-class passengers. Several other first-class passengers also have fares above 200, suggesting that high fares can be legitimate. The 512.3292 ticket appears only once in this dataset, so its exact pricing reason cannot be determined from the available columns.*

Decision: Keep the value; don't delete it without evidence that it is an error.